File: libgdx_ninja_runner/core/src/com/nopalsoft/ninjarunner/objects/Pet.java
package com.nopalsoft.ninjarunner.objects;

import com.badlogic.gdx.math.Vector2;
import com.badlogic.gdx.physics.box2d.Body;

public class Pet {
    public final static int STATE_NORMAL = 0;

    public final static float SPEED = 5f;
    public static final float RADIUS = .25f;
    public final PetType petType;
    public final float drawWidth;
    public final float drawHeight;
    public final float dashDrawWidth;
    public final float dashDrawHeight;
    public final Vector2 position;
    final public Vector2 targetPosition;
    public int state;
    public Vector2 velocity;
    public float stateTime;

    public Pet(float x, float y, PetType petType) {
        this.petType = petType;
        position = new Vector2(x, y);
        targetPosition = new Vector2(x, y);
        velocity = new Vector2();
        state = STATE_NORMAL;
        stateTime = 0;

        switch (petType) {
            case PINK_BIRD:
                drawWidth = .73f;
                drawHeight = .66f;
                dashDrawWidth = 2.36f;
                dashDrawHeight = 1.25f;
                break;
            default:
            case BOMB:
                drawWidth = dashDrawWidth = .52f;
                drawHeight = dashDrawHeight = .64f;
                break;

        }

    }

    public void update(Body body, float delta, float targetX, float targetY) {
        position.x = body.getPosition().x;
        position.y = body.getPosition().y;

        targetPosition.set(targetX, targetY);

        velocity = body.getLinearVelocity();
        velocity.set(targetPosition).sub(position).scl(SPEED);
        body.setLinearVelocity(velocity);
        stateTime += delta;
    }


    public void updateStateTime(float delta) {
        stateTime += delta;

    }

    public enum PetType {
        PINK_BIRD, BOMB

    }

}

File: libgdx_ninja_runner/core/src/com/nopalsoft/ninjarunner/objects/Pet.kt
package com.nopalsoft.ninjarunner.objects

import com.badlogic.gdx.math.Vector2
import com.badlogic.gdx.physics.box2d.Body

/**
 * The pet that the user will have and follows them through the game.
 */
class Pet(x: Float, y: Float, val petType: PetType) {


    val position = Vector2(x, y)
    private val targetPosition = Vector2(x, y)
    private var velocity = Vector2()
    var stateTime = 0f
    var drawWidth = 0f
    var drawHeight = 0f
    var dashDrawWidth = 0f
    var dashDrawHeight = 0f

    init {
        when (petType) {
            PetType.PINK_BIRD -> {
                drawWidth = .73f
                drawHeight = .66f
                dashDrawWidth = 2.36f
                dashDrawHeight = 1.25f
            }

            PetType.BOMB -> {
                dashDrawWidth = .52f
                drawWidth = dashDrawWidth
                dashDrawHeight = .64f
                drawHeight = dashDrawHeight
            }

        }
    }

    fun update(body: Body, delta: Float, targetX: Float, targetY: Float) {
        position.x = body.position.x
        position.y = body.position.y

        targetPosition.set(targetX, targetY)

        velocity = body.linearVelocity
        velocity.set(targetPosition).sub(position).scl(SPEED)
        body.linearVelocity = velocity
        stateTime += delta
    }

    fun updateStateTime(delta: Float) {
        stateTime += delta
    }


    companion object {

        const val SPEED = 5f
        const val RADIUS = .25f
    }

    enum class PetType {
        PINK_BIRD, BOMB
    }
}